### PR TITLE
 #551 Added JSON.INGEST support for HTTP

### LINKED ISSUE #551: Add Support for JSON.INGEST Command
Add support for a new command, JSON.INGEST allows users to set a value based on a dynamically generated key. The key will be generated using a key prefix and a unique identifier provided by the user. 

The key prefix will be specified in the URL’s query parameters.

### Command Overview:

**Command**: JSON.INGEST
**Syntax**: JSON.INGEST <key_prefix> <JSON_value>
**Example**: JSON.INGEST gitmr_ {"value": "oval"}

### Details:

This command functions similarly to the SET command, but instead of accepting a complete key, it only takes a key prefix.

The whole key is generated by appending a unique identifier to the provided key prefix.
The generated key is then used to store the provided JSON value.

### Use Case:

This command will allow integration with webhooks where we don't have control over the payload passed to us.

For example, the webhook URL could be: 

https://<host>/JSON.INGEST?key_prefix=pref_

In this scenario, the webhook requests the specified URL and the JSON.INGEST command dynamically generates the key based on the key_prefix provided in the query parameter and stores the associated JSON payload.

### Tasks:

- Implement the JSON.INGEST command in the codebase.
- Create comprehensive documentation in DiceDB Docs for the JSON.INGEST command, including usage examples, parameter descriptions, and potential use cases.
- Write unit tests to validate the behaviour of this command.

### PR DESCRIPTION
**Pull request by kushal0511-not**
Added JSON.INGEST support.
closes #551.

image

Screenshot from 2024-09-17 22-45-55

Screenshot from 2024-09-15 14-16-26

What is remaining: 

- unit tests
- documentation

### CODE REVIEW

**Comment by kushal0511-not** [comment]
@JyotinderSingh please review this draft PR.
 If changes look good, I will add unit test cases for command.

File: internal/eval/eval.go [hunk 1]
```diff
@@ -1126,6 +1127,34 @@ func evalJSONARRAPPEND(args []string, store *dstore.Store) []byte {
...
+	uniqueID := uuid.New()
+	generatedKey := keyPrefix + uniqueID.String()
+
+	var jsonValue interface{}
+	if err := sonic.UnmarshalString(jsonStr, &jsonValue); err != nil {
+		return diceerrors.NewErrWithFormattedMessage("invalid JSON: %v", err.Error())
+	}
+
+	newObj := store.NewObj(jsonValue, -1, object.ObjTypeJSON, object.ObjEncodingJSON)
+	store.Put(generatedKey, newObj)
```
Comment (lucifercr07) [inline internal/eval/eval.go:1153]: Please correct me if I'm missing something, can we not use a call to `JSONSET` here?

File: internal/eval/eval.go [hunk 2]
```diff
@@ -1126,6 +1127,34 @@ func evalJSONARRAPPEND(args []string, store *dstore.Store) []byte {
...
+// Returns response.RespOK if the JSON value is successfully stored
+func evalJSONINGEST(args []string, store *dstore.Store) []byte {
+	if len(args) < 2 {
+		return diceerrors.NewErrArity("JSON.INGEST")
+	}
+
+	keyPrefix := args[0]
+	jsonStr := args[1]
+
+	uniqueID := uuid.New()
```
Comment (lucifercr07) [inline internal/eval/eval.go:1144]: Not sure if we required a 128 bits uuid for this. May be we can use some other lib here.

**Review by lucifercr07 — CHANGES_REQUESTED** [review]

**Review by lucifercr07 — COMMENTED** [review]
LGTM, please make the PR ready for review to approve.

**Comment by kushal0511-not** [comment]
@lucifercr07 Thanks for reviewing.
I have made it ready for review.
I have one question , do i need to include any unit testcase for this command(HTTP protocol )  under this issue? if yes, what should be the most suitable location for that.
Or  test cases will be covered in https://github.com/DiceDB/dice/issues/558 or https://github.com/DiceDB/dice/issues/552

**Comment by kushal0511-not** [comment]
@lucifercr07 Thanks for reviewing.
Resolved conflicts.

**Comment by pratikpandey21** [comment]
Small recommendation, but I think it might make sense to return the generated ID as response, rather than OK. @kushal0511-not

**Comment by kushal0511-not** [comment]
Changes done @pratikpandey21.
Please review  it.

**Review by lucifercr07 — APPROVED** [review]

**Pull request merged**